### PR TITLE
v32 removals

### PR DESCRIPTION
**Pull request by albi3ro**
This PR completes the removals for:

* `qchem.molecular_hamiltonian` keyword arguments `grouping_type` and `grouping_method`
* `zyz_decompposition` and `xyx_decomposition`
* `LieAlgebraOptimizer`
* `Operation.base_name`
* `QuantumScript.name`
* `qml.math.reduced_dm`
* `qml.specs` keys `num_operations`, `num_used_wires`, `gate_types`, `gate_sizes` and `depth`

The `do_queue` keyword is not removed in this PR, as that will change a lot of code.

### CODE REVIEW

**Comment by albi3ro** [comment]
[sc-38961] [sc-38962] [sc-38963]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -21,4 +43,5 @@
 
 This release contains contributions from (in alphabetical order):
 
-Matthew Silverman
+Christina Lee
```
Comment (timmysilv) [inline doc/releases/changelog-dev.md:46]:
```suggestion
Christina Lee,
```

File: pennylane/transforms/decompositions/single_qubit_unitary.py [hunk 2]
```diff
@@ -46,53 +45,6 @@ def _convert_to_su2(U, return_global_phase=False):
...
-    [Rot(-0.24209529417800013, 1.14938178234275, 1.7330581433950871, wires=[0])]
-    """
-    warnings.warn(
-        "The zyz_decomposition function is deprecated and will be removed soon. Use"
-        " one_qubit_decomposition() with the keyword rotations='ZYZ'"
-    )
-    return _zyz_decomposition_old(U=U, wire=wire)
-
-
 def _zyz_decomposition_old(U, wire):
```
Comment (timmysilv) [inline pennylane/transforms/decompositions/single_qubit_unitary.py:48]: if we're keeping this, perhaps we shouldn't keep the `_old` suffix?

**Review by timmysilv — APPROVED** [review]

**Review by mudit2812 — APPROVED** [review]

File: pennylane/transforms/decompositions/single_qubit_unitary.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/transforms/decompositions/single_qubit_unitary.py]: renamed to `_rot_decomposition`.

**Pull request merged**